### ISSUE TITLE
Delete an entry (entry detail)

### ISSUE THREAD

**Issue by sandysage** [mozilla-lockbox/lockbox-ios#175]
When I notice I have duplicate or inaccurate entries, I want the ability to delete the credential entirely from my database.

## Requirements
* Should provide the ability to delete from the entry detail view #937 #938
* Should have a disclaimer that I have to confirm when deleting a record #939 
* Should have ability to cancel the delete request #939 
* Should not have the ability to undo (can consider as a future request)
* Should display toast to confirm deletion on returning to entry list #940

NOTE: public release should include both edit and delete because flows are intrinsically tied

Future requirement (not in scope): ability to delete entries from the list view #197 

## Visual Design
**InVision**
Entry Detail - Edit: https://mozilla.invisionapp.com/share/BQEUSO0S2#/289206367_Entry_Detail_-_Edit
Entry Detail - Confirm Delete: https://mozilla.invisionapp.com/share/BQEUSO0S2#/289206366_Entry_Detail_-_Edit_-_Confirm_Delete

**Zeplin**
Entry Detail - Edit: https://app.zeplin.io/project/5a4e9ff8032d0923200a2635/screen/5c6dc07d1380fe4628cb6cc4
Entry Detail - Confirm Delete: https://app.zeplin.io/project/5a4e9ff8032d0923200a2635/screen/5c6dc07d959ef19a21360492

## Epic Acceptance Criteria
All requirements are addressed by engineering stories.

**Comment by changecourse** [comment]
We need to strongly consider supporting "swipe to delete" functionality via the entry list in parallel with this story...

**Comment by sashei** [comment]
@changecourse one thing that occurred to me while breaking all these out: do we want / need to have a dedicated design for iPad?
